x/bank/keeper/msg_server.go

```diff
@@ -181,3 +181,56 @@ func (k msgServer) SetSendEnabled(goCtx context.Context, msg *types.MsgSetSendEn
 
 	return &types.MsgSetSendEnabledResponse{}, nil
 }
+
+func (k msgServer) Burn(goCtx context.Context, msg *types.MsgBurn) (*types.MsgBurnResponse, error) {
+	var (
+		from []byte
+		err  error
+	)
+
+	var coins sdk.Coins
+	for _, coin := range msg.Amount {
+		coins = append(coins, sdk.NewCoin(coin.Denom, coin.Amount))
+	}
```

Comment:
Why do we need to recreate the coins slice?

Reply:
> Mainly tried to avoid using gogo annotations

Asking to learn, is this a thing we're generally striving towards?

Reply:
mainly in prep for protov2/pulsar but we dot really have a plan for how to get there today